src/handleAction.js

```diff
@@ -7,19 +7,15 @@ export default function handleAction(type, reducers, defaultState) {
     ? type.toString()
     : type;
 
+  const [nextReducer, throwReducer] = isFunction(reducers)
+    ? [reducers, reducers]
+    : [reducers.next, reducers.throw];
+
   return (state = defaultState, action) => {
     // If action type does not match, return previous state
     if (action.type !== typeValue) return state;
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
-
-    // If function is passed instead of map, use as reducer
-    if (isFunction(reducers)) {
-      reducers.next = reducers.throw = reducers;
-    }
-
-    // Otherwise, assume an action map was passed
-    const reducer = reducers[handlerKey];
+    const reducer = action.error === true ? throwReducer : nextReducer;
 
     return isFunction(reducer)
       ? reducer(state, action)
```

Comment:
I think this is a case of a breaking change, but it's also the library doing what the documentation says

>If a single reducer is passed, it is used to handle both normal actions and failed actions. (A failed action is analogous to a rejected promise.) You can use this form if you know a certain type of action will never fail, like the increment example above.
>
> Otherwise, you can specify separate reducers for next() and throw(). This API is inspired by the ES6 generator interface.

So it seems it doesn't make sense to not pass `throw` and just pass `next` in the object form, since the former case would handle that.

For your implementation, why not do an `isFunction` check instead of checking `undefined`? 

Also, you could ensure that you can dot into `reducers` first before checking the types of `reducers.throw` and `reducers.next`. [This would have thrown errors previously](https://github.com/acdlite/redux-actions/blob/master/src/handleAction.js#L21), so it's not a breaking change, we'd just be being more explicit about why it failed here.